pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -90,6 +90,23 @@ type PipelineResult struct {
 	Value string `json:"value"`
 }
 
+type Metadata struct {
```

Comment:
Thanks for making this change! Would you be open to renaming it something like `PipelineTaskMetadata`? The name is a bit vague as it is.

The name of the type can be changed in the future without breaking our YAML-focused definition of backward-compatibility AIUI, so this might not be a big deal.

Reply:
yup this can be changed in future without breaking anyone. I had named it `PipelineTaskMetadata` and then changed it to `Metadata` thinking this could be generic and used under any other section in future 💭

Reply:
Done, since its in the `pipeline_types.go`, would be better to name it `PipelineTaskMetadata` 👍